backend/alembic/versions/0034_virtual_collections_db_view.py

```diff
@@ -0,0 +1,306 @@
+                            '$[*]' COLUMNS (franchise VARCHAR(255) PATH '$')
+                        ) j
+                    WHERE
+                        JSON_EXTRACT(igdb_metadata, '$.franchises') IS NOT NULL
+                ),
+                collections AS (
+                    SELECT
+                        r.id as rom_id,
+                        r.path_cover_s as path_cover_s,
+                        r.path_cover_l as path_cover_l,
+                        CONCAT(j.collection) as collection_name,
+                        'collection' as collection_type
+                    FROM
+                        roms r
+                        CROSS JOIN JSON_TABLE(
+                            JSON_EXTRACT(igdb_metadata, '$.collections'),
+                            '$[*]' COLUMNS (collection VARCHAR(255) PATH '$')
+                        ) j
+                    WHERE
+                        JSON_EXTRACT(igdb_metadata, '$.collections') IS NOT NULL
+                ),
+                modes AS (
+                    SELECT
+                        r.id as rom_id,
+                        r.path_cover_s as path_cover_s,
+                        r.path_cover_l as path_cover_l,
+                        CONCAT(j.mode) as collection_name,
+                        'mode' as collection_type
+                    FROM
+                        roms r
+                        CROSS JOIN JSON_TABLE(
+                            JSON_EXTRACT(igdb_metadata, '$.game_modes'),
+                            '$[*]' COLUMNS (mode VARCHAR(255) PATH '$')
+                        ) j
+                    WHERE
+                        JSON_EXTRACT(igdb_metadata, '$.game_modes') IS NOT NULL
+                ),
+                companies AS (
+                    SELECT
+                        r.id as rom_id,
+                        r.path_cover_s as path_cover_s,
+                        r.path_cover_l as path_cover_l,
+                        CONCAT(j.company) as collection_name,
+                        'company' as collection_type
+                    FROM
+                        roms r
+                        CROSS JOIN JSON_TABLE(
+                            JSON_EXTRACT(igdb_metadata, '$.companies'),
+                            '$[*]' COLUMNS (company VARCHAR(255) PATH '$')
+                        ) j
+                    WHERE
+                        JSON_EXTRACT(igdb_metadata, '$.companies') IS NOT NULL
+                )
+                SELECT
+                    collection_name as name,
+                    collection_type as type,
+                    CONCAT('Autogenerated ', collection_name, ' collection') AS description,
+                    NOW() AS created_at,
+                    NOW() AS updated_at,
+                    JSON_ARRAYAGG(DISTINCT rom_id) as rom_ids,
+                    JSON_ARRAYAGG(DISTINCT path_cover_s) as path_covers_s,
+                    JSON_ARRAYAGG(DISTINCT path_cover_l) as path_covers_l
+                FROM
```

Comment:
Should these three fields always match their orderings? e.g. Second rom_id corresponds with second element for the cover paths in the other arrays.

If that's the case, I think we should enforce sorting using something like:
```sql
                    JSON_ARRAYAGG(DISTINCT rom_id ORDER BY rom_id) as rom_ids,
                    JSON_ARRAYAGG(DISTINCT path_cover_s ORDER BY rom_id) as path_covers_s,
                    JSON_ARRAYAGG(DISTINCT path_cover_l ORDER BY rom_id) as path_covers_l
```

Reply:
the magic here is that it'll always be in order, even without `ORDER BY`!
<img width="213" alt="Screenshot 2025-02-09 at 8 36 17 PM" src="https://github.com/user-attachments/assets/449ce36c-76e8-43fe-b9dd-deab29ba1b81" />